presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -156,8 +166,81 @@ public void testPlanningFailure()
         assertFailedQuery("SELECT * FROM mock.default.tests_table", "Throw from apply projection");
     }
 
+    @Test
+    public void testAbortedWhileWaitingForResources()
+            throws Exception
+    {
+        Session mySession = Session.builder(getSession())
+                .setSystemProperty("required_workers_count", "17")
+                .setSystemProperty("required_workers_max_wait_time", "10ms")
+                .build();
+        assertFailedQuery(mySession, "SELECT * FROM tpch.sf1.nation", "Insufficient active worker nodes. Waited 10.00ms for at least 17 workers, but only 1 workers are active");
+    }
+
+    @Test
+    public void testKilledWhileWaitingForResources()
+            throws Exception
+    {
+        String testQueryMarker = "test_query_id_" + ThreadLocalRandom.current().nextInt(100_000_000);
```

Comment:
`[UUID.] randomUUID().toString().replace("-", "")`

Reply:
I also updated same code in `TestKillQuery`